src/lepton/jpgcoder.cc

```diff
@@ -1443,7 +1443,7 @@ std::string uniq_filename(std::string filename) {
 
 std::string postfix_uniq(const std::string &filename, const char * ext) {
     std::string::size_type where =filename.find_last_of("./\\");
-    if (where == std::string::npos || filename[where] != '.') {
+    if (where == std::string::npos || filename[where] != '.' || filename.substr(where) != ".jpg") {
```

Comment:
What is it needed for?

Reply:
Originally, lepton replaced ANY filename extension with ".lep". It was weird when we are going to process files like "007.012345 007.56789" - both output files were "007.lep"

This patch replaces with ".lep" only "jpg" filename extension, otherwise ".lep" is added to the original filename.

Reply:
I don't implemented similar mechanism for reverse conversation, though, since all my lepton files have .lep extension...